### PR TITLE
fix: remove unnecessary permissions from Appointment and Appointment Booking Settings

### PR DESCRIPTION
**Pull request by DaizyModi**

### CODE REVIEW

File: erpnext/crm/doctype/appointment/appointment.py [hunk 1]
```diff
@@ -240,3 +242,14 @@ def _get_employee_from_user(user):
 	if employee_docname:
 		return frappe.get_doc("Employee", employee_docname)
 	return None
+
+
+def add_agent_assignment(args):
```
Comment (barredterra) [inline erpnext/crm/doctype/appointment/appointment.py]:
```suggestion
def assign_agents(doctype: str, name: str, agents: list[str]) -> None:
```

Maybe this signature would improve readability a bit.

File: erpnext/www/book_appointment/index.py [hunk 2]
```diff
@@ -90,23 +94,27 @@ def get_available_slots_between(query_start_time, query_end_time, settings):
 
 @frappe.whitelist(allow_guest=True)
 def create_appointment(date, time, tz, contact):
-	format_string = "%Y-%m-%d %H:%M:%S"
-	scheduled_time = datetime.datetime.strptime(date + " " + time, format_string)
+	contact = json.loads(contact)
+	datetime_obj = datetime.datetime.strptime(date + " " + time, "%Y-%m-%d %H:%M:%S")
```
Comment (barredterra) [inline erpnext/www/book_appointment/index.py]:
```suggestion
	from frappe.utils.data import get_datetime

	contact = json.loads(contact)
	datetime_obj = get_datetime(date + " " + time)
```

We can use frappe's `get_datetime` util here

Comment (DaizyModi) [reply erpnext/www/book_appointment/index.py]: Yes. We will refactor it in another PR.

**Comment by deepeshgarg007** [comment]
Didn't see the squash label :sweat_smile:

**Pull request merged**